appgate/types.py

```diff
@@ -260,7 +260,10 @@ def id(self) -> str:
 
     @property
     def tags(self) -> FrozenSet[str]:
-        return self.value.tags
+        try:
+            return self.value.tags
```

Comment:
I guess singleton entities can't be tagged?
Might be worth adding a comment on the attribute error.